examples/ecs/prepaid-instance/variables.tf

```
@@ -0,0 +1,213 @@
+# Variable definitions for authentication
+variable "region_name" {
+  description = "The region where the ECS instance is located"
+  type        = string
+}
+
+variable "access_key" {
+  description = "The access key of the IAM user"
+  type        = string
+}
+
+variable "secret_key" {
+  description = "The secret key of the IAM user"
+  type        = string
+}
+
+# Variable definitions for resources/data sources
+variable "availability_zone" {
+  description = "The availability zone to which the ECS instance belongs"
+  type        = string
+  default     = ""
+}
+
+variable "instance_flavor_id" {
+  description = "The flavor ID of the ECS instance"
+  type        = string
+  default     = ""
+}
```

```suggestion
variable "instance_flavor_id" {
  description = "The flavor ID of the ECS instance"
  type        = string
  default     = ""
  nullable    = false
}
```
Do not allow the null value input if you are using condition expression in main.tf